https://www.openwall.com/lists/musl/2023/11/06/3
https://inbox.vuxu.org/musl/20231106113336.3664-2-ncopa@alpinelinux.org/T/#u
https://sourceware.org/bugzilla/show_bug.cgi?id=31034
https://bugs.gentoo.org/916857

From mboxrd@z Thu Jan  1 00:00:00 1970
X-Spam-Checker-Version: SpamAssassin 3.4.4 (2020-01-24) on inbox.vuxu.org
X-Spam-Level: 
X-Spam-Status: No, score=-3.1 required=5.0 tests=DKIM_INVALID,DKIM_SIGNED,
	MAILING_LIST_MULTI,RCVD_IN_DNSWL_MED,RCVD_IN_MSPIKE_H3,
	RCVD_IN_MSPIKE_WL,T_SCC_BODY_TEXT_LINE autolearn=ham
	autolearn_force=no version=3.4.4
Received: (qmail 5179 invoked from network); 6 Nov 2023 11:46:34 -0000
Received: from second.openwall.net (193.110.157.125)
  by inbox.vuxu.org with ESMTPUTF8; 6 Nov 2023 11:46:34 -0000
Received: (qmail 30570 invoked by uid 550); 6 Nov 2023 11:46:29 -0000
Mailing-List: contact musl-help@lists.openwall.com; run by ezmlm
Precedence: bulk
List-Post: <mailto:musl@lists.openwall.com>
List-Help: <mailto:musl-help@lists.openwall.com>
List-Unsubscribe: <mailto:musl-unsubscribe@lists.openwall.com>
List-Subscribe: <mailto:musl-subscribe@lists.openwall.com>
List-ID: <musl.lists.openwall.com>
Reply-To: musl@lists.openwall.com
Received: (qmail 30538 invoked from network); 6 Nov 2023 11:46:29 -0000
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=alpinelinux.org;
	s=smtp; t=1699271177;
	h=from:from:reply-to:subject:subject:date:date:message-id:message-id:
	 to:to:cc:cc:mime-version:mime-version:
	 content-transfer-encoding:content-transfer-encoding;
	bh=73HDLjg72r1JGckDGbEyPxYrYL7dC7MB3gMwy/yp7hc=;
	b=pSGCs/DrFDbs9eEA89un578pZbyzpmTw81QGH7xK4ZAAkYiXx1ysaXlsllwxGd076F+plw
	kE1QbGVndutc+ieeUOiHomF4O8IP4AqO/8xCy52LlYmnhMTcxgoXD/GWHfVcXmIgFb+8Uc
	jvgM9nXFOXceFSlHLLOwJBQFE2dyBrU=
From: Natanael Copa <ncopa@alpinelinux.org>
To: musl@lists.openwall.com
Cc: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon,  6 Nov 2023 12:33:37 +0100
Message-ID: <20231106113336.3664-2-ncopa@alpinelinux.org>
X-Mailer: git-send-email 2.42.1
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Subject: [musl] [PATCH] elf.h: add typedefs for Elf*_Relr

Add typedefs for Elf32_Relr and Elf64_Relr as a follow-up to commit
d32dadd60efb (ldso: support DT_RELR relative relocation format)

---
This fixes build of iproute2 with elfutils 0.190, which assumes that
Elf*_Relr are typedef'ed when SHT_RELR is defined.

ref: https://sourceware.org/git/?p=elfutils.git;a=commit;h=39f2c500542f69c2f1a13fd0ae4eaa5778d2ed8d
ref: https://sourceware.org/bugzilla/show_bug.cgi?id=31034

 include/elf.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/elf.h b/include/elf.h
index 23f2c4bc..aa186d9d 100644
--- a/include/elf.h
+++ b/include/elf.h
@@ -32,6 +32,9 @@ typedef uint16_t Elf64_Section;
 typedef Elf32_Half Elf32_Versym;
 typedef Elf64_Half Elf64_Versym;
 
+typedef Elf32_Word Elf32_Relr;
+typedef Elf64_Xword Elf64_Relr;
+
 #define EI_NIDENT (16)
 
 typedef struct {
-- 
2.42.1
